### PR TITLE
[OLGA] Change services for geoIP, fix bugs, refactor code

### PR DESCRIPTION
**Pull request by arsentur**
**Description:**
Changed services for getting geodata. Fix bugs with sending data to OLGA. Refactor code.
Write documentation for installing edx-platform and OLGA locally and connects them.

**Youtrack:**
https://youtrack.raccoongang.com/issue/OLGA-55

**Documentation:**
https://docs.google.com/document/d/1MrmGMW_d6wqN_KDT17aS6fBebgX0TwDoKJ280CN-gbQ/edit#

### CODE REVIEW

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

**Comment by svic** [comment]
retest this PR please

File: openedx/core/djangoapps/edx_global_analytics/utils/utilities.py [hunk 1]
```diff
@@ -79,46 +79,17 @@ def get_previous_month_start_and_end_dates():
 
 def get_coordinates_by_ip():
     """
-    Gather coordinates by server IP address with FreeGeoIP service.
+    Gather coordinates by server IP address with ip-api service.
```
Comment (idegtiarov) [inline openedx/core/djangoapps/edx_global_analytics/utils/utilities.py]: Title in the multiple doc strings should be separated from the other text by the empty line.

File: openedx/core/djangoapps/edx_global_analytics/utils/utilities.py [hunk 2]
```diff
@@ -79,46 +79,17 @@ def get_previous_month_start_and_end_dates():
...
-        latitude, longitude = ip_data.json()['latitude'], ip_data.json()['longitude']
-        return latitude, longitude
+    latitude, longitude = '', ''
 
-    except requests.RequestException as error:
-        logger.exception(error.message)
-        return '', ''
+    ip_data = requests.get('http://ip-api.com/json')
 
+    if ip_data.status_code == 200:
```
Comment (idegtiarov) [inline openedx/core/djangoapps/edx_global_analytics/utils/utilities.py]: should `ip_data.json()` be also checked before we start taking values by the keys?

File: openedx/core/djangoapps/edx_global_analytics/tasks.py [hunk 3]
```diff
@@ -117,16 +117,17 @@ def collect_stats():
...
-        latitude, longitude = platform_coordinates(platform_city_name)
+        latitude, longitude = get_coordinates_by_ip()
 
         students_per_country = enthusiast_level_statistics_bunch()
 
         data.update({
             'latitude': latitude,
             'longitude': longitude,
             'platform_name': platform_name,
+            'platform_cty_name': platform_city_name,
```
Comment (idegtiarov) [inline openedx/core/djangoapps/edx_global_analytics/tasks.py]: typo: s/platform_cty_name/platform_city_name

File: openedx/core/djangoapps/edx_global_analytics/tests/test_platform_coordinates.py [hunk 4]
```diff
@@ -13,43 +13,28 @@
...
 class TestPlatformCoordinates(unittest.TestCase):
     """
-    Tests for platform coordinates methods, that gather latitude and longitude.
+    Tests for platform coordinates method, that gather latitude and longitude.
     """
 
     def tests_sending_requests(self, mock_request):
         """
-        Tests to prove that methods send request to needed corresponding URLs.
+        Test to prove that method send request to needed corresponding URL.
```
Comment (idegtiarov) [inline openedx/core/djangoapps/edx_global_analytics/tests/test_platform_coordinates.py]: Typo: s/send/sends/

File: openedx/core/djangoapps/edx_global_analytics/tests/test_utils.py [hunk 5]
```diff
@@ -10,11 +10,11 @@
 from django.db.models import Q
 from django_countries.fields import Country
 
+from openedx.core.djangoapps.edx_global_analytics.utils import fetch_instance_information, cache_instance_data
```
Comment (idegtiarov) [inline openedx/core/djangoapps/edx_global_analytics/tests/test_utils.py]: Please improve import order

**Pull request merged**